File: spotipie/src/domain/Song.ts
import { Artist } from './Artist';

export class Song {
	constructor(private artists: Artist[], private title: string, private albumName: string, private image: string){}

	public static of(artists: Artist[], title: string, albumName: string, image: string): Song {
		return new Song(artists, title, albumName, image);
	}

	public getTitle(): string {
		return this.title;
	}

	public getArtists(): Artist[] {
		return this.artists;
	}

	public getAlbumName(): string {
		return this.albumName;
	}

	public getImage(): string {
		return this.image;
	}
}

File: spotipieBack/src/main/java/com/spotipie/domain/entity/song/Song.java
package com.spotipie.domain.entity.song;

import java.util.List;

import com.spotipie.domain.assertions.AssertMissingMandatoryValue;
import com.spotipie.domain.entity.Image.Image;
import com.spotipie.domain.entity.artist.Artist;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@ToString
@EqualsAndHashCode
@Getter
public class Song {
  private final List<Artist> artists;
  private final String title;
  private final String albumName;
  private final List<Image> urlImages;

  @Builder
  public Song(List<Artist> artists, String title, String albumName, List<Image> urlImages) {
    AssertMissingMandatoryValue.verify("artists", artists);
    AssertMissingMandatoryValue.verify("title", title);
    AssertMissingMandatoryValue.verify("albumName", albumName);
    AssertMissingMandatoryValue.verify("urlImages", urlImages);

    this.artists = artists;
    this.title = title;
    this.albumName = albumName;
    this.urlImages = urlImages;
  }
}
